### PR TITLE
Detail page

### PR DESCRIPTION
**Pull request by daiterranova**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
     }
 ```
 
-Using the `filter` method for arrays, we go through the watchList array and returned it that list except the stock that was selected for deletion.
-We render this new filtered list passing the new array to the watchList variable to `setWatchList`
+Using the `filter` method, we go through the watchList array and returned it excepting of the selected stock.
```
Comment (reynico) [inline README.md]:
```suggestion
Using the `filter` method, we go through the watchList array and return it except of the selected stock.
```

File: README.md [hunk 2]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
...
+We set the value of the `watchList` variable to this new filtered list.
+
+### useNavigate Hook
+
+When the user clicks on a row, we want to re-directed to the detail of the stock.
```
Comment (reynico) [inline README.md]:
```suggestion
When the user clicks on a row, we want to be re-directed to the detail of the stock.
```

File: README.md [hunk 3]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
...
+
+Inside of the `StockDetailPage` component:
+
+First, import `useParams` hook to extract the **symbol**:
+
+```
+const { symbol } = useParams()
+```
+
+After, we implement a **useEffect** hook to fetch the data:
```
Comment (reynico) [inline README.md]:
```suggestion
Then, we implement a **useEffect** hook to fetch the data:
```

File: README.md [hunk 4]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
...
+
+- Create a date object ``` const date = new Date() ```
+- Get the current time ```const currentTime = Math.floor(date.getTime() / 1000)````
+- Establish the neccesary conditions for contemplate the weekend days when there's no new quotes.
```
Comment (reynico) [inline README.md]:
```suggestion
- Establish the necessary conditions to contemplate the weekend days when there are no new quotes. 
```

File: README.md [hunk 5]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
...
+                oneDay = currentTime - 2 * 24 * 60 * 60
+            }
+            else if (day === 0) { //sunday
+                oneDay = currentTime - 3 * 24 * 60 * 60
+            }
+            else {
+                oneDay = currentTime - 24 * 60 * 60
+            }
+```
+- Using **Promise.all** and taking into account the finnhub endpoint for the [stock candles](https://finnhub.io/docs/api/stock-candles), we get the data that we need hitting to the **/stock/candle's endpoint** as the first parameter, and as the second one, the object params with the required arguments (symbol,from,to,resolution). We fetch three request for obtain the historical data of a day, a week and a year.
```
Comment (reynico) [inline README.md]:
```suggestion
- Using **Promise.all** and taking into account the finnhub endpoint for the [stock candles](https://finnhub.io/docs/api/stock-candles), we get the data that we need hitting to the **/stock/candle's endpoint** as the first parameter, and as the second one, the object params with the required arguments (symbol,from,to,resolution). We fetch three requests to obtain the historical data of a day, a week and a year.
```

File: README.md [hunk 6]
```diff
@@ -452,5 +452,111 @@ For now, we only create the function to eliminate an stock from the list:
...
+                            from: oneYear,
+                            to: currentTime,
+                            resolution: "M"
+                        }
+                    })
+                ])
+```
+#### Formatting data
+
+Now  that we have the responses we need, we stored it on a new state variable `chartData`, and inside of the useEffect we set the variable with an object with the properties day,week and year, and each one of them call the **formatData** function that receives the data property for every request made it.
```
Comment (reynico) [inline README.md]:
```suggestion
Now  that we have the responses we need, we store them on a new state variable `chartData`, and inside of the useEffect we set the variable with an object with the properties day,week and year, and each one of them calls the **formatData** function that receives the data property for every request made it.
```

File: src/apis/finnHub.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import axios from "axios";
 
-let TOKEN = "cgd20g1r01qum7u5quagcgd20g1r01qum7u5qub0"
+let TOKEN = "cggt9g1r01qv7vi3iom0cggt9g1r01qv7vi3iomg"
```
Comment (reynico) [inline src/apis/finnHub.js:3]: while this doesn't suppose a security problem, bear in mind that tokens, secrets, API keys and so on shouldn't be stored on Github as plain text.

**Review by reynico — CHANGES_REQUESTED** [review]

**Comment by reynico** [comment]
I'd just address the wording on the README file. Other than that code looks good and it's good to go.

File: src/apis/finnHub.js (hunk 7 above)
Comment (daiterranova) [reply src/apis/finnHub.js:3]: mmm good point. I'm not sure how to not show my personal API key and make the project public

Comment (daiterranova) [reply src/apis/finnHub.js:3]: Maybe I can use the api key from the tutorial's project

Comment (reynico) [reply src/apis/finnHub.js:3]: A simple way to handle this problem is to use environment variables, https://www.twilio.com/blog/working-with-environment-variables-in-node-js-html

that way you could define `TOKEN` as follows:

```js
let TOKEN = process.env.TOKEN
```

then in your computer's console you do:

```bash
export TOKEN="cgd..."
npm run dev
```

**Pull request merged**